src/store/RelayQueryTracker.js

```diff
@@ -60,8 +60,16 @@ class RelayQueryTracker {
     const {isMerged, trackedNodes} = trackedNodesByID;
     if (!isMerged) {
       const trackedChildren = [];
+      let wrapperNode = null;
       trackedNodes.forEach(trackedQuery => {
-        trackedChildren.push(...trackedQuery.getChildren());
+        wrapperNode = new RelayQuery.Fragment.build(
```

Comment:
Also I think `getType` *may* return null (I'm on mobile otherwise I'd check - look at RelayQuery.js). If so, then only add the wrapping fragment if there is a type.